### ISSUE TITLE
Operator spelling bikeshedding

### ISSUE THREAD

**Issue by js-choi** [tabatkins/proposal-call-operator#2]
`…::(…)` is fine but a tad bit verbose. We should consider other spellings, too.

~~`…^(…)`~~
`…!(…)`
`…~(…)` (although this is also taken by [partial function application](https://github.com/tc39/proposal-partial-application))
`…@(…)`
`…#(…)`

**Comment by tabatkins** [comment]
I don't want to overlap with PFA (particularly since I'd want this to *work with* PFA, as the combo becomes a bind-this operator), but yeah, I have no opinion on what character to use.

We do still have to avoid binary ops like `^`, since `a^(b)` is valid code today. But all the other suggestions in your list work. `!`, in particular, is somewhat cute paired with optional-call, tho they aren't really related so that might actually be a bad thing.

**Comment by ljharb** [comment]
tbh i like the `@` since you're calling a function "at" an object - ie, it implies location.

**Comment by js-choi** [comment]
Yes, good point on `^`; I had forgotten.

I agree that `!` is pretty good. I don’t think analogy with `?.` would be a big problem. There are several programming languages that give to “bang” function calls (with `!`) meanings unrelated to function calls with `?`—whether by syntax or by convention. Examples include Elixir, Julia, Rust, and Ruby.

**Comment by ljharb** [comment]
The meaning is quite different though. In TS, it means "i assert this is non-nullish". In Ruby, it means "this is dangerous", which in Rails also means "throw an exception if it fails, instead of returning a sentinel" .

**Comment by js-choi** [comment]
@ljharb: Yes, sorry, I’m not saying that those other languages’ bang function calls would share the same meaning as ES call-this. 

Instead, I’m saying that those other languages share dissimilarity in meaning between their `!` function calls and their `?` function calls. With those existing precedents, I don’t think that confusion in JavaScript between call-this (with `!`) and optional calls (with `?.`) would be a big problem.

I do also like the analogy between `@` and the “location” of a function call.
